### PR TITLE
Pocket Casts: Add Pocket Casts block for internal dev purposes.

### PR DESCRIPTION
**Pull request by yansern**
#### Changes proposed in this Pull Request:
This PR adds a Pocket Casts block variation for **internal dev purposes**.

This allows the Pocket Casts team to test this block against their oEmbed endpoint which is still under development.

#### Jetpack product discussion
pdDOJh-bf-p2 & pdeCcb-dg-p2

#### Does this pull request change what data or activity we track or use?
No

#### Testing instructions:

1. Apply patch D77986-code.
2. Add `add_filter( 'jetpack_block_editor_pocket_casts_feature', '__return_true' );` to your config file. (See D77986-code for more details).
3. Open up block editor.
4. Add "Pocket Casts" block to your editor.
5. Pasting a pocket cast link will not work at the moment as the oEmbed service is still under development.
 
image

image

### CODE REVIEW

**Comment by ivan-ottinger** [comment]
Is there anything else apart from adding `define( 'JETPACK_BETA_BLOCKS', true );` to `tools/docker/wordpress/wp-config.php` needed to make the block display?

I couldn't make the block display on neither sandboxed site, nor WoA yet. I guess I am missing something. 🤔

File: projects/plugins/jetpack/changelog/add-pocket-casts-dev-block [hunk 1]
```diff
@@ -0,0 +1,4 @@
+Significance: minor
+Type: other
+
+Add pocket casts dev block (for internal use only).
```
Comment (ivan-ottinger) [inline projects/plugins/jetpack/changelog/add-pocket-casts-dev-block:4]: Maybe we could add a "category" to specify where what is the change related to, e.g.:

```suggestion
Blocks: Add Pocket Casts dev block (for internal use only)
```

**Comment by yansern** [comment]
> Is there anything else apart from adding `define( 'JETPACK_BETA_BLOCKS', true );` to `tools/docker/wordpress/wp-config.php` needed to make the block display?

That should be the only thing needed.

Just covering all bases. Double-check if Jetpack plugin is activated. Also, try re-run `jetpack install --all`.

> I couldn't make the block display on neither sandboxed site, nor WoA yet. I guess I am missing something. 🤔

I personally have not tried on the sandboxed site, just following what was written in:https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/extensions/README.md#beta-extensions.

Hope there's nothing I missed there. (I'll double-check again later.)

File: projects/plugins/jetpack/changelog/add-pocket-casts-dev-block (hunk 1 above)
Comment (yansern) [reply projects/plugins/jetpack/changelog/add-pocket-casts-dev-block]: Thanks!

**Review by jeherve — COMMENTED** [review]
> If we decide to continue this as a core-embed block variation, we will migrate over to extensions/extended-blocks/core-embed

That would be my vote, personally. It would seem like a better fit there.

> The reason it is placed under the extensions/block folder is to leverage the JETPACK_BETA_BLOCKS flag which is otherwise impossible if placed under extensions/extended-blocks/core-embed.

**Comment by yansern** [comment]
> Maybe we could do something similar here?

Thank you @jeherve. This was what I was looking for and allows me to jump straight to using block variation! PR https://github.com/Automattic/jetpack/pull/23741 closed and changes moved back in here.

> Would it be possible to keep the Jetpack color for the block icon, like we do for the other blocks and embeds, even those where the logo usually has a distinctive color, like Facebook.

Icon color now inherits from Jetpack.

This PR is ready to test again. (cc @ivan-ottinger).

Testing instructions updated. For now just add `add_filter( 'jetpack_block_editor_pocket_casts_feature', '__return_true' );` to the sandbox config for the Pocket Casts block to show up. 

_Another phab patch that returns true when user is an a12n is planned but doesn't block this PR._

**Review by ivan-ottinger — APPROVED** [review]
Looks good to me. 👍🏼 Just left one minor comment.

Pocket Casts block variation is available:
Markup on 2022-04-04 at 10:41:38

File: projects/plugins/jetpack/class.jetpack-gutenberg.php [hunk 2]
```diff
@@ -701,6 +701,16 @@ public static function enqueue_block_editor_assets() {
...
 					'republicize_enabled'       => apply_filters( 'jetpack_block_editor_republicize_feature', true ),
+					/**
+					 * Enable PocketCasts block variation in block editor context.
+					 *
+					 * @module pocketcasts
+					 *
+					 * @since 12.9.0
+					 *
+					 * @param bool true Enable PocketCasts block variation in block editor context. Defaults to false.
+					 */
```
[the remark this answers is not in the document]
Comment (yansern) [reply projects/plugins/jetpack/class.jetpack-gutenberg.php]: Thank you!

**Review by jeherve — CHANGES_REQUESTED** [review]
Should be good, short of 2 tiny changes.

**Review by jeherve — APPROVED** [review]

**Comment by yansern** [comment]
Changeset ID: f512cacb6b42fdde3a06ef3cea12f8082d3ba05d

Also related: D78101-code

**Pull request merged**